packages/core/content-releases/admin/src/services/axios.ts

```diff
@@ -0,0 +1,91 @@
+import { getFetchClient } from '@strapi/helper-plugin';
+
+import type { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios';
+
+/* -------------------------------------------------------------------------------------------------
+ * Axios data
+ * -----------------------------------------------------------------------------------------------*/
+export interface QueryArguments<TSend> {
+  url: string;
+  method: 'PUT' | 'GET' | 'POST' | 'DELETE';
+  data?: TSend;
+  config?: AxiosRequestConfig<TSend>;
+}
+
+const axiosBaseQuery = async <TData = any, TSend = any>({
+  url,
+  method,
+  data,
+  config,
+}: QueryArguments<TSend>) => {
+  try {
+    const { get, post, del, put } = getFetchClient();
+
+    if (method === 'POST') {
+      const result = await post<TData, AxiosResponse<TData>, TSend>(url, data, config);
+      return { data: result.data };
+    }
+
+    if (method === 'DELETE') {
+      const result = await del<TData, AxiosResponse<TData>, TSend>(url, config);
+      return { data: result.data };
+    }
+
+    if (method === 'PUT') {
+      const result = await put<TData, AxiosResponse<TData>, TSend>(url, data, config);
+      return { data: result.data };
+    }
+
+    /**
+     * Default is GET.
+     */
+    const result = await get<TData, AxiosResponse<TData>, TSend>(url, config);
+    return { data: result.data };
+  } catch (error) {
+    if (isErrorAxiosError(error)) {
+      /**
+       * Handle error of type AxiosError
+       *
+       * This format mimics what we want from an AxiosError which is what the
+       * rest of the app works with, except this format is "serializable" since
+       * it goes into the redux store.
+       *
+       * NOTE – passing the whole response will highlight this "serializability" issue.
+       */
+      return {
+        error: {
+          status: error.response?.status,
+          code: error.code,
+          response: {
+            data: error.response?.data,
+          },
+        },
+      };
+    } else {
+      // Handle error of type unknown
+      throw new Error('An error occured');
+    }
```

Comment:
@remidej not sure if this is what you had in mind...

I can't just return error of type unknown when it's not an axioserror or I end up only with error type unknown on the response. For example with createRelease.